lib/python/qmk/cli/doctor.py

```diff
@@ -20,6 +24,137 @@ def _udev_rule(vid, pid=None):
+    """
+    ok = True
+    udev_dir = Path("/etc/udev/rules.d/")
+    desired_rules = {
+        'dfu': {_udev_rule("03eb", "2ff4"), _udev_rule("03eb", "2ffb"), _udev_rule("03eb", "2ff0")},
+        'tmk': {_udev_rule("feed")},
+        'input_club': {_udev_rule("1c11")},
+        'stm32': {_udev_rule("1eaf", "0003"), _udev_rule("0483", "df11")},
+        'caterina': {'ATTRS{idVendor}=="2a03", ENV{ID_MM_DEVICE_IGNORE}="1"', 'ATTRS{idVendor}=="2341", ENV{ID_MM_DEVICE_IGNORE}="1"'},
+    }
+
+    if udev_dir.exists():
+        udev_rules = [str(rule_file) for rule_file in udev_dir.glob('*.rules')]
+        current_rules = set()
+
+        # Collect all rules from the config files
+        for rule_file in udev_rules:
+            with open(rule_file, "r") as fd:
+                for line in fd.readlines():
+                    line = line.strip()
+                    if not line.startswith("#") and len(line):
+                        current_rules.add(line)
+
+        # Check if the desired rules are among the currently present rules
+        for bootloader, rules in desired_rules.items():
+            if not rules.issubset(current_rules):
+                # If the rules for catalina are not present, check if ModemManager is running
+                if bootloader == "caterina":
+                    if check_modem_manager():
+                        ok = False
+                        cli.log.warn("{bg_yellow}Detected ModemManager without udev rules. Please either disable it or set the appropriate udev rules if you are using a Pro Micro.")
+                else:
+                    cli.log.warn("{bg_yellow}Missing udev rules for '%s' boards. You'll need to use `sudo` in order to flash them.", bootloader)
+
+    return ok
+
+
+def check_modem_manager():
+    """Returns True if ModemManager is running.
+    """
+    if shutil.which("systemctl"):
+        mm_check = subprocess.run(["systemctl", "--quiet", "is-active", "ModemManager.service"], timeout=10)
+        if mm_check.returncode == 0:
+            return True
+
+    else:
+        cli.log.warn("Can't find systemctl to check for ModemManager.")
+
+
+def is_executable(command):
+    """Returns True if command exists and can be executed.
+    """
+    # Make sure the command is in the path.
+    res = shutil.which(command)
+    if res is None:
+        cli.log.error("{fg_red}Can't find %s in your path.", command)
+        return False
+
+    # Make sure the command can be executed
+    check = subprocess.run([command, '--version'], stdout=subprocess.PIPE, stderr=subprocess.PIPE, timeout=5)
+    if check.returncode in [0, 1]:  # Older versions of dfu-programmer exit 1
+        cli.log.debug('Found {fg_cyan}%s', command)
+        return True
+
+    cli.log.error("{fg_red}Can't run `%s --version`", command)
+    return False
+
+
+def os_test_linux():
+    """Run the Linux specific tests.
+    """
+    cli.log.info("Detected {fg_cyan}Linux.")
+    ok = True
+
+    if not check_udev_rules:
```

Comment:
```suggestion
    if not check_udev_rules():
```
Besides this, looks ready to be merged.